src/HealthGPS.Console/model_parser.cpp

```diff
@@ -184,32 +184,43 @@ load_dynamic_risk_model_definition(const host::poco::json &opt) {
 std::shared_ptr<EnergyBalanceModelDefinition>
 load_newebm_risk_model_definition(const host::poco::json &opt) {
     MEASURE_FUNCTION();
-    std::vector<core::Identifier> nutrient_list;
-    std::map<core::Identifier, std::map<core::Identifier, double>> nutrient_equations;
-
-    // Save nutrient identities.
-    auto nutrient_strings = opt["Nutrients"].get<std::vector<std::string>>();
-    for (const std::string &nutrient_str : nutrient_strings) {
-        nutrient_list.emplace_back(nutrient_str);
+    std::unordered_map<core::Identifier, double> energy_equation;
+    std::unordered_map<core::Identifier, std::map<core::Identifier, double>> nutrient_equations;
+    std::unordered_map<core::Gender, std::vector<double>> age_mean_height;
+
+    // Save nutrient -> energy equation.
+    for (const auto &nutrient : opt["Nutrients"]) {
+        auto nutrient_str = nutrient["Name"].get<std::string>();
+        auto nutrient_key = core::Identifier{nutrient_str};
+        auto nutrient_energy = nutrient["Energy"].get<double>();
+        energy_equation[nutrient_key] = nutrient_energy;
     }
 
     // Save food -> nutrient equations.
     for (const auto &food : opt["Foods"]) {
-        auto food_ident = core::Identifier{food["Name"].get<std::string>()};
-        auto food_nutrient_strings = food["Nutrients"].get<std::map<std::string, double>>();
+        auto food_str = food["Name"].get<std::string>();
+        auto food_key = core::Identifier{food_str};
```

Comment:
Again, `food_str` could be moved.

As this seems to be a fairly common pattern in this file, how about making a helper function, called `get_identifier()` or something? It would take in the JSON object and return a `core::Identifier`.

Reply:
Actually, I've got a better suggestion.

The JSON library lets you serialise/deserialise arbitrary types if you provide helpers: https://json.nlohmann.me/features/arbitrary_types/#basic-usage

So you should be able to just put this little shim somewhere:

```cpp
namespace hgps::core {
void from_json(const host::poco::json& j, Identifier& id) {
    id = Identifier(j.get<std::string>());
}
} // namespace hgps::core
```

Then you can do:

```cpp
auto food_key = food["Name"].get<core::Identifier>();
```

Reply:
Ah, good shout. Forgot about that.